definitions/mobile.d.ts

```diff
@@ -6,49 +6,92 @@ declare module Mobile {
-		getVersion(): string;
-		getVendor(): string;
-		getPlatform(): string;
+		deviceInfo: Mobile.IDeviceInfo;
+		applicationManager: Mobile.IDeviceApplicationManager;
+		fileSystem: Mobile.IDeviceFileSystem;
 		deploy(packageFile: string, packageName: string): IFuture<void>;
-		sync(localToDevicePaths: ILocalToDevicePathData[], appIdentifier: IAppIdentifier, liveSyncUrl: string): IFuture<void>;
-		sync(localToDevicePaths: ILocalToDevicePathData[], appIdentifier: IAppIdentifier, liveSyncUrl: string, options: ISyncOptions): IFuture<void>;
 		openDeviceLogStream(): void;
-		runApplication(applicationId: string): IFuture<void>;
-		uninstallApplication(applicationId: string): IFuture<void>;
-		listFiles(devicePath: string): IFuture<void>;
-		getFile(deviceFilePath: string): IFuture<void>;
-		putFile(localFilePath: string, deviceFilePath: string): IFuture<void>;
+	}
+	
+	interface IAndroidDevice extends IDevice {
+		adb: Mobile.IAndroidDebugBridge;		
+		sendBroadcastToDevice(action: string, extras: IStringDictionary): IFuture<number>;
 	}
 
-	interface IAppIdentifier {
+	interface IIOSDevice extends IDevice {
+		startService(serviceName: string): number;
+		mountImage(): IFuture<void>;
+		tryToExecuteFunction<TResult>(func: () => TResult): TResult;
+	}
+	
+	interface IDeviceAppDataBase {
 		appIdentifier: string;
-		deviceProjectPath: string;
-		liveSyncFormat: string;
-		encodeLiveSyncHostUri(hostUri: string): string;
-		isLiveSyncSupported(device: any): IFuture<boolean>;
-		getLiveSyncNotSupportedError(device: any): string;
 	}
 
-	interface IAndroidDevice extends IDevice {
-		debug(packageFile: string, packageName: string, debuggerSetup?: any): IFuture<void>;
+	interface IDeviceAppData extends IDeviceAppDataBase {
+		deviceProjectRootPath: string;
+		isLiveSyncSupported(device: Mobile.IDevice): IFuture<boolean>;
+	}
+	
+	interface IDeviceAppDataFactory {
+		create(appIdentifier: string, platform: string): Mobile.IDeviceAppData;
+	}
+	
+	interface IDeviceAppDataFactoryRule {
+		vanilla: any;
+		companion?: any;
+	}
+	
+	interface IDeviceAppDataProvider {
+		createFactoryRules(): IDictionary<Mobile.IDeviceAppDataFactoryRule>;
 	}
 
-	interface IIOSDevice extends IDevice {
-		startService(serviceName: string): number;
+	interface IAndroidLiveSyncService {
+		liveSyncCommands: any;
+		livesync(appIdentifier: string, liveSyncRoot: string, commands: string[]): IFuture<void>;
+		createCommandsFileOnDevice(commandsFileDevicePath: string, commands: string[]): IFuture<void>;
 	}
 
 	interface ILogcatHelper {
-		start(deviceIdentifier: string, adbPath: string): any;
+		start(deviceIdentifier: string): any;
 	}
 
 	interface ILogcatPrinter {
 		print(line: string): void;
 	}
+	
+	interface IDeviceApplicationManager {
+		getInstalledApplications(): IFuture<string[]>;
+		installApplication(packageFilePath: string): IFuture<void>;
+		uninstallApplication(appIdentifier: string): IFuture<void>; 
+		runApplication(appIdentifier: string): IFuture<void>;
```

Comment:
Consider either renaming this method to `startApplication` or rename the `stopApplication` to something else. The main issue is that when you have such pair of methods they should have common prefixes: `start`/`stop`, `begin`/`end`, etc.